pytorch_lightning/utilities/argparse_utils.py

```diff
@@ -91,6 +91,7 @@ def get_init_arguments_and_types(cls) -> List[Tuple[str, Tuple, Any]]:
          ('profiler',
           (<class 'pytorch_lightning.profiler.profilers.BaseProfiler'>,
            <class 'bool'>,
+           <class 'str'>,
```

Comment:
```suggestion
           <class 'str'>,
```

I think this is why tests fail. Two options:
1. include bool in the type annotation of Trainer
2. remove the bool here. 

Either way, the signature of Trainer must match what is listed here.

Reply:
I went with option no. 1, see [commit](https://github.com/PyTorchLightning/pytorch-lightning/pull/3656/commits/d7b2e7449d883a425fe0f7ee8da7199dc45eae84). 

`bool` needs to stay in the `profiler` type in `trainer.py` because , `Trainer.add_argparse_args(parent_parser=parser)` deduces from the types of the `Trainer`'s `profiler` parameter what type the argument added to `ArgParser` should have. If `bool` is not in the signature of the `profiler` parameter, the type of the argument added to `ArgParser` is `str`.

Then if CLI args like "--profiler True", the value of the parsed arg would be `"True"` instead of `True` and `Trainer` would raise an exception complaining that only `"simple"` and `"advanced"` are acceptable `str` values.

Reply:
@awaelchli could you please take a look at this PR? If you are satisfied with the changes, I think it can be merged :)

Reply:
yes, this seems to be resolved now. 
the PR does not have a milestone, so I don't know if this will go into v1.0. 
need to ask @williamFalcon @edenafek

Reply:
Okay, thanks I haven't realized that.